genealogy/geneweb/Portfile

```
@@ -1,109 +1,127 @@
-PortSystem 1.0
-
-name			geneweb
-version			5.02
-revision		3
-categories		genealogy
-maintainers		nomaintainer
-description		Sophisticated Genealogy System 
-long_description	GeneWeb is sophisticated genealogy software with a \
-			web interface.  It provides efficient relationship and consanguinity \
-			computing, support for very large database (i.e., hundreds of thousands \
-			of entries), and more.
-
-homepage		http://cristal.inria.fr/~ddr/GeneWeb
-master_sites		http://cristal.inria.fr/~ddr/GeneWeb/ftp/Src \
-			ftp://ftp.inria.fr/INRIA/Projects/cristal/${name}/Src
-distfiles		${name}-${version}.tgz
-checksums		md5 e8c6c789a9dd2f9155bf6eefa8786d01 \
-			sha1 e040496b3ce106cc7a508360cd0e4268bcb3e7b1 \
-			rmd160 fd284630e11957cf1265bfce9207e46d449a83ec
-platforms		darwin
-
-depends_build		port:ocaml \
-			port:camlp5
-
-patchfiles		patch-5.02-1
-
-use_parallel_build	no
-
-destroot.target		distrib
-
-set gwuser		${name}
-set gwgroup		${name}
-set logdir		${prefix}/var/log
-set langdir		${prefix}/share/${name}/gw
-set dbdir		${prefix}/var/db/${name}
-
-startupitem.create	yes
-startupitem.init        HOME=${prefix}/share/${name}
-startupitem.start	"su ${gwuser} -c \"cd ${prefix}/share/${name} && ./gw/gwd -bd${dbdir} -hd${langdir} -dd${langdir} -langen -log${logdir}/${name}.log -daemon\""
-startupitem.pidfile	none
-
-platform darwin {
-	if {${configure.build_arch} eq "x86_64" || ${configure.build_arch} eq "ppc64"} {
-		patchfiles-append patch-64-bit
-	}
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+name                geneweb
+version             7.1-beta
+revision            0
+github.setup        ${name} ${name} ${version} v
+categories          genealogy
+maintainers         {pguyot @pguyot} openmaintainer
+license             GPL-2
+
+description         Sophisticated Genealogy System
+long_description    GeneWeb is sophisticated genealogy software with a \
+                    web interface.  It provides efficient relationship and consanguinity \
+                    computing, support for very large database (i.e., hundreds of thousands \
+                    of entries), and more.
+
+homepage            https://geneweb.tuxfamily.org/wiki/GeneWeb
+extract.suffix      .zip
+checksums           rmd160  3745f4476378fb90cd21f628ae6a2e93dd0dae21 \
+                    sha256  527bc5a59e378cad144449800d214c641d775f5269716f3c40e68c5de87eab9a \
+                    size    14465775
```

This port does not download a zip file, and we would not want it to, so `extract.suffix zip` should be removed. Setting it merely causes this port to download a tar.gz file that is misnamed to have a zip extension. (If you wanted a zip file, which you don't, you would do it by setting `use_zip yes` which will change not only the file suffix but also the extract commands.)

By default, the github portgroup downloads an automatically-generated tar.gz file from the legacy "tarball" endpoint. The use of this legacy endpoint is not recommended. Use the newer "archive" endpoint instead, by specifying `github.tarball_from archive`. The checksums will need to be changed as a result.

```suggestion
github.tarball_from archive
checksums           rmd160  7fa38db901155a79369b50368450db71db5b48f1 \
                    sha256  bfb2d45e3e1bd2b702ed5a46c8c1d1319bcf23d75d1c1ce771bfb489a5589073 \
                    size    14465662
```